themes/startwords/assets/js/notes.js

```diff
@@ -1,170 +1,243 @@
+            // end notes may already be visible.
+            if (this.endnotes.classList.contains('endnotes')) {
+                // hide end notes so the note can be shown as a contextual note
+                this.endnotes.classList.remove('endnotes');
+                // set a flag that endnotes need to be re-shown
+                ContextualNotes.showEndnotesOnClose = true;
+                // temporarily disable hide on scroll, since
+                // hiding the endnotes changes document length
+                // and can trigger scrolling
+                this.disableHideOnScroll(2);
+            }
+
+            this.positionContextualNote(note);
+            // get back reference link to override behavior
+            var backref = note.getElementsByClassName('footnote-backref').item(0);
+            backref.addEventListener('click', ContextualNotes.noteCloseLinkHandler);
         }
-    }, 500);
+    }
 
-    // create observable to set endnotes class on footnotes ol when visible
-    var observer = new IntersectionObserver(endnoteIntersectionCallback);
-    var endnotes = document.querySelector('.footnotes');
-    if (endnotes) observer.observe(endnotes);
-});
+    static closeNote(note) {
+        note = (typeof note !== 'undefined') ?  note : document.getElementById(location.hash.slice(1));
+        if (note) {
+            // change the location hash, deselecting the note
+            // Set to non-existent id so note is no longer targeted,
+            // but document does not scroll
+            location.hash = '#-';
+            // unbind the close link handler
+            var backref = note.getElementsByClassName('footnote-backref').item(0);
+            backref.removeEventListener('click', ContextualNotes.noteCloseLinkHandler);
+
+            // if endnotes were hidden to show this note, redisplay them
+            if (ContextualNotes.showEndnotesOnClose) {
+                ContextualNotes.showEndnotes()
+                ContextualNotes.showEndnotesOnClose = false;
+            }
+        }
+    }
 
-window.addEventListener('hashchange', function() {
-    setupNote();
-}, false);
+    static noteCloseLinkHandler() {
+        // prevent default behavior to avoid jarring scroll to reference
+        // Check that the parent note is still selected; otherwise behave normally
+
+        // backref may be inside a paragraph OR directly inside the li
+        var note = this.parentNode;
+        if (note.localName != 'li') {
+            note = note.parentNode;
+        }
 
-/* on scroll, if a footnote is currently targeted, unselect */
-window.addEventListener('scroll', function() {
-    if (location.hash.startsWith('#fn:') && hideOnScroll) {
-        // close the note and unbind close link handler
-        closeNote(getCurrentNote());
+        if (location.hash == '#' + note.getAttribute('id')) {
+            event.preventDefault();
+            event.stopPropagation();
```

Comment:
linter is giving me a warning I don't totally understand here, but I think it's to the effect of:
- this reference to `event` would be `undefined`...
- but it works, because of a (deprecated) spec that allows you to access the event anyway
- you can resolve it (i think) by just explicitly listing `event` as an argument to this handler

Reply:
tried that and it seems to work.